packages/itwinui-react/src/core/Tile/Tile.tsx

```diff
@@ -147,239 +105,463 @@ type TileProps = {
-      onClick,
-      ...rest
-    } = props;
-
-    const [isMenuVisible, setIsMenuVisible] = React.useState(false);
-    const showMenu = React.useCallback(() => setIsMenuVisible(true), []);
-    const hideMenu = React.useCallback(() => setIsMenuVisible(false), []);
-    const [localActionable, setLocalActionable] =
-      React.useState(isActionableProp);
-
-    const isActionable = isActionableProp ?? localActionable;
-
-    const tileName = (
-      <Box className='iui-tile-name'>
-        <TitleIcon
-          isLoading={isLoading}
-          isSelected={isSelected}
-          isNew={isNew}
-          status={status}
-        />
-
-        <Box as='span' className='iui-tile-name-label'>
-          {isActionable && onClick ? (
-            <LinkAction
-              as='button'
-              onClick={!isDisabled ? onClick : undefined}
-              aria-disabled={isDisabled}
-            >
-              {name}
-            </LinkAction>
-          ) : (
-            name
-          )}
-        </Box>
-      </Box>
-    );
+const TileComponent = React.forwardRef((props, ref) => {
+  const {
+    className,
+    status,
+    variant,
+    isNew,
+    isSelected,
+    isLoading,
+    isActionable: isActionableProp,
+    isDisabled,
+    onClick,
+    ...rest
+  } = props;
+  const [localActionable, setLocalActionable] =
+    React.useState(isActionableProp);
+  const isActionable = isActionableProp ?? localActionable;
+
+  return (
+    <TileContext.Provider
+      value={{
+        status,
+        variant,
+        isNew,
+        isSelected,
+        isLoading,
+        isActionable,
+        isDisabled,
+        setActionable: setLocalActionable,
+        onClick,
+      }}
+    >
+      <Box
+        className={cx(
+          'iui-tile',
+          {
+            'iui-folder': variant === 'folder',
+            'iui-new': isNew,
+            'iui-selected': isSelected,
+            'iui-actionable': isActionable,
+            [`iui-${status}`]: !!status,
+            'iui-loading': isLoading,
+          },
+          className,
+        )}
+        aria-disabled={isDisabled}
+        ref={ref}
+        {...rest}
+      />
+    </TileContext.Provider>
+  );
+}) as PolymorphicForwardRefComponent<'div', TileOwnProps>;
+TileComponent.displayName = 'Tile';
+
+// ----------------------------------------------------------------------------
+// Tile.Action component
+
+const TileAction = React.forwardRef((props, forwardedRef) => {
+  const { setActionable } = useSafeContext(TileContext);
+  React.useEffect(() => {
+    if (!supportsHas()) {
+      setActionable(true);
+    }
+  }, [setActionable]);
+
+  return <LinkAction ref={forwardedRef} {...props} />;
+}) as PolymorphicForwardRefComponent<'a'>;
+TileAction.displayName = 'Tile.Action';
+
+// ----------------------------------------------------------------------------
+// Tile.ThumbnailArea component
+
+const TileThumbnailArea = polymorphic.div('iui-tile-thumbnail');
```

Comment:
You can replace `polymorphic.div(` with `polymorphic(` since `div` is the default. This change can be made across the entire file

Reply:
oooh thanks! I updated all of the similar instances.